### PR TITLE
VOTE-441: Add husky support. Include git hooks for git-secrets to pre…

### PR DESCRIPTION
**Pull request by daniel-crowder**
…vent committing AWS secrets.

## Jira ticket

[VOTE-441](https://cm-jira.usa.gov/browse/VOTE-441)

## Description

Install husky and git-secrets to enable the use of commit hooks to prevent AWS secrets from being committed.

### Pre-deploy

1. Insert any required steps to take before merging and running post-deploy.

### Post-deploy

1. Insert any required steps to take before beginning test.

### QA/Test

1. Ensure that you npm installed locally, outside of any Docker container.
2. Install git-secrets locally - https://github.com/awslabs/git-secrets#installing-git-secrets
3. Confirm `git-secrets` executes.
4. Once installed, run `npm install` which will install Husky for the project.
5. Confirm that you receive the message `husky - Git hooks installed` after Step 3.
6. Create or alter a file to include a random AWS key. An example key is available here: https://awsteele.com/blog/2020/09/26/aws-access-key-format.html
Example: demo.txt with key, ASIA...................A
7. Run the following
`git add {test_file.txt}`
`git commit -m"I should not see this in logs."` - The commit should be blocked and you should receive a message showing where a secret was found
8. Use `git restore --staged <file>` to unstage your changes so you can delete the test key.

TIP: You can always issue the following command for a scan
`git secrets --scan` - You should receive a message showing where a secret was found

## Checklist for the Developer

- [x] A link to the JIRA ticket has been included above.
- [x] No merge conflicts exist with the target branch.
- [x] Automated tests have passed on this PR.
- [x] A reviewer has been designated.
- [x] Deployment and testing steps have been documented above, if applicable.

## Checklist for the Peer Reviewers

- [ ] The file changes are relevant to the task objective.
- [ ] Code is readable and includes appropriate commenting.
- [ ] Code standards and best practices are followed.
- [ ] QA/Test steps were successfully completed, if applicable.
- [ ] Applicable logs are free of errors.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "devDependencies": {
+    "husky": "^8.0.0"
+  },
+  "scripts": {
+    "prepare": "husky install"
```
Comment (clmedders) [inline package.json:6]: Can i suggest that we add a command here something like `"secretScan": "git secrets --scan"`, this way we can add it to the pre-commit hook... 
```suggestion
    "prepare": "husky install",
    "secretScan": "git secrets --scan"
```

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    exit 1
+fi
+
+# Initialise git-secrets configuration
+git-secrets --register-aws > /dev/null
+
+echo "Running git-secrets..."
+# Scans all files that are about to be committed.
+git-secrets --pre_commit_hook -- "$@"
+
```
Comment (clmedders) [inline .husky/pre-commit:17]: You can add the command here and it will run for every commit after it is stagged
```suggestion
npm run secretScan
```

**Review by clmedders — DISMISSED** [review]
looks good... was able to run through the commands and was not able to commit something with the AWS key in it. 

just a thought... this seems like a targeted process I made a few suggestions to have it run on every commit that way in case someone is new and does not know the process or even if someone like my self were to have a key and didn't know I should not push it up it will catch it no matter what and give use that extra layer of protection.

If we go down that path we will need to make sure we document it some where about the other install and let the other devs know as well

File: .husky/pre-commit (hunk 2 above)
Comment (daniel-crowder) [reply .husky/pre-commit:17]: This is redundant.

**Comment by rayestrada** [comment]
Since we have implement secret push protections in Github security settings this is no longer needed.

**Pull request closed without merging**